### ISSUE TITLE
flow name no longer stored for errors

### ISSUE THREAD

**Issue by joecrean** [marklogic/marklogic-data-hub#1845]
### Welcome Hub enthusiast!

Let's get the most out of this ticket.

#### The issue

Short description of the problem:

What behavior are you expecting?

#### Tech details

Which Operating System are you using?

Which version of MarkLogic are you using?

9.0.6-1

Which version of the Data Hub Framework are you using?

4.1.1

#### Description

In my application one specific harmonisation flow has functionality to reprocess errors out of the TRACING/JOBS db. Because every flow writes error records it is necessary to be able to query error records for a given flow (with the aim of reprocessing them by that flow). In previous versions (<4.x) of the DHF the error data was written to the TRACING db and the flow name was stored in the JOBS db. Now the TRACING db has been removed and the data which was previously written to the TRACING db is now written to JOBS... the old data including the flow name is not merged in with this and is simply no longer written meaning that there is no longer any way to determine flow information and query error records by flow.

**Comment by joecrean** [comment]
we could add the flow name very simply to the error record in `rfc:writer-error-trace` in the source file `/data/4/impl/traxce-lib.xqy`

**Comment by aebadirad** [comment]
If you're willing to branch 4.x-develop and create a fix and PR it back, we'll review it and include it if it solves your use case.

**Comment by joecrean** [comment]
@aebadirad 
will do

**Comment by joecrean** [comment]
The function `rfc:get-flow` returns a `hub:flow` element which currently looks like the following...I have anonymised some of the details however clearly we need the name element here. 
```
<flow xmlns="http://marklogic.com/data-hub">
  <name>Foo</name>
  <entity>MyEntity</entity>
  <type>harmonize</type>
 <data-format>xml</data-format>
 <code-format>xqy</code-format>
 <collector code-format="xqy" module="/entities/MyEntity/harmonize/Foo/collector.xqy"/>
 <main code-format="xqy" module="/entities/MyEntity/harmonize/Foo/main.xqy"/>
</flow>
```

**Comment by joecrean** [comment]
@aebadirad 
i have added the changes in a local branch off `4.x-develop` but get a 403 when i push.. can you authorise me?

**Comment by aebadirad** [comment]
Fixed for version 4.2.1 @kkanthet let's include this issue# on change log
